### ISSUE TITLE
Alternative and MonadPlus instances in the module Choose

### ISSUE THREAD

**Issue by SkorikGG** [suhailshergill/extensible-effects#67]
for using `guard`.
```
instance Member Choose r => Alternative (Eff r) where
    empty = mzero'
    (<|>) = mplus'

instance  Member Choose r => MonadPlus (Eff r)
```

**Comment by suhailshergill** [comment]
@SkorikGG apologies for the delay, i've been bouncing from one conference to the next.
Do you have patches to submit that i can review?

**Comment by suhailshergill** [comment]
this is partially resolved by #71 
`NdetEff` now has the desired `Alternative` and `MonadPlus` instances.

To add the desired instances in `Choose` requires refactoring the tests.
